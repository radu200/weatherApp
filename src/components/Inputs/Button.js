import React from "react";
import PropTypes from 'prop-types';

const Button = ({
  name,
  type,
  className,
  handleClick
}) => {
  return (
    <button  data-test-id="button-search" className={className} onClick={handleClick} type={type}>
      {name}
    </button>
  );
};

Button.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  className:PropTypes.string,
  handleClick: PropTypes.func
}
export default Button;
