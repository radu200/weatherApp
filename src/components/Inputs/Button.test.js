import React from 'react'
import Button from './Button'
import { shallow} from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'


describe('Button component', () => {
 
   const props = {
      name:'search',
      type:"submit",
      className:'someClass'
   }


   let wrap 
   beforeEach(() => {
      wrap = shallow(<Button {...props}/>)
   })

   it('should render button with given props', () => {
        expect(shallowToJson(wrap)).toMatchSnapshot()
   })

    
    it('should call click event', () => {
       const mockFn = jest.fn() 
       const component = shallow(<Button handleClick={mockFn} />)
       component.find('button').simulate('click')
       expect(mockFn).toHaveBeenCalled();

    })
})
