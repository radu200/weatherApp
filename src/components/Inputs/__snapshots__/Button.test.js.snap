// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render button with given props 1`] = `
<button
  className="someClass"
  data-test-id="button-search"
  type="submit"
>
  search
</button>
`;
